### DISCUSSION TITLE
Importing OTEL metrics and tracing data

### DISCUSSION THREAD

**Discussion by rj-au** [datalust/seq-tickets#2019] [Help] [answered]
Currently Seq has an endpoint to receive OTEL log data, but we'd like to also accept OTEL metric and trace data.

First of all, is it a good idea to ingesting OTEL metric and trace data into Seq?
We think we will be able to setup adequate alerts and dashboard once this data is in Seq.

Could we export from an OTEL collector to Seq the metric and trace data?
For OTEL Logs we can already configure the exporter `otlphttp/seq` to the endpoint: ".../ingest/otlp"

Also, looks like trace support is coming soon: https://github.com/datalust/seq-tickets/issues/1986

**Comment by liammclennan** [comment, marked as answer]
Hi @richard-stafflink 

See the [roadmap](https://datalust.co/roadmap). 

Trace support will be in the next Seq release. Metrics are planned, but not scheduled. 

Seq does / will support OTLP so you can use the OTEL collector, or send directly to Seq.

**Comment by BrianVallelunga** [comment]
Any update on OTEL metrics? Think they'll make it in this year?

**Reply by nblumhardt** [reply]
Not this year, unfortunately. Scale-out query execution and HA clustering are currently mid-flight, and will be our major focus for at least the next six months or so.

We're actually using metrics in Seq quite heavily though, internally, to monitor the in-development HA implementation. We're doing this using OTLP and a simple adapter that combines and pivots metric samples into wide-ish structured log events.

This works pretty well for the small volume of data we're collecting when testing (just a few source servers, short retention window) but as you'd expect, it won't scale to the kinds of metric volumes real-world deployments will need.

All of this is helping to build a picture of how metric data should fit into Seq's data model and query experience, plus we're getting some insights into the way metrics can drive dashboarding and alerting in that model. Our aim is to hit the ground running once the metrics roadmap item does move to "in progress" - although the back-end work required for efficient time-series storage is very significant, so there's no way to pin down even a rough timeline, at this stage.
